enhancements/installer/single-node-installation-bootstrap-in-place.md

```diff
@@ -0,0 +1,309 @@
+
+We will add a new `post-reboot` service for approving the kubelet and the node Certificate Signing Requests.
+This service will also cleanup the bootstrap static pods resources once the OCP control plane is ready.
+Since we start with a liveCD, the bootstrap services (`bootkube`, `approve-csr`, etc.), `/etc` and `/opt/openshift` "garbage" are written to the ephemiral filesystem of the liveCD, and not to the node's real filesystem.
+The files that we need to delete are under:
+`/etc/kubernetes/bootstrap-secrets` and `/etc/kubernetes/bootstrap-configs`
+These files are required for the bootstrap control plane to start before it is replaced by the control plane operators.
+Once the OCP control plane static pods are deployed we can delete the files as they are no longer required. 
+
+### Initial Proof-of-Concept
+
+User flow
+1. Generate bootstrap ignition using the OpenShift installer.
+2. Embed this Ignition to an RHCOS liveCD.
+3. Boot a machine with this liveCD.
+
+This POC uses the following services for mitigating some gaps:
+- `patch.service` for allowing single node installation. it won't be required once [single-node production deployment](https://github.com/openshift/enhancements/pull/560/files) is implemented.
+- `post_reboot.service` for approving the node CSR and bootstrap static pods resources cleanup post reboot.
+
+Steps to try it out:
+- Clone the installer branch: `iBIP_4_6` from https://github.com/eranco74/installer.git
+- Build the installer (`TAGS=libvirt hack/build.sh`)
+- Add your ssh key and pull secret to the ``./install-config.yaml`
+- Generate ignition - `make generate`
+- Set up networking - `make network` (provides DNS for `Cluster name: test-cluster, Base DNS: redhat.com`)
+- Download rhcos image - `make embed` (download RHCOS liveCD and embed the bootstrap Ignition)
+- Spin up a VM with the the liveCD - `make start-iso`
+- Monitor the progress using `make ssh` and `journalctl -f -u bootkube.service` or `kubectl --kubeconfig ./mydir/auth/kubeconfig get clusterversion`
+
+
+### Risks and Mitigations
+
+*What are the risks of this proposal and how do we mitigate. Think broadly. For
+example, consider both security and how this will impact the larger OKD
+ecosystem.*
+
+*How will security be reviewed and by whom? How will UX be reviewed and by whom?*
+
+## Design Details
+
+### Open Questions
+
+2. Bootable installation artifact?, perhaps the installer should produce the liveCD?
+3. How will the user specify costum configuration, such as installation disk, static IPs? 
+4. Number of revisions for the control plane - do we want to make changes to the bootstrap static pods to make them closer to the final ones?
+6. What do we do with the CRDs that are generated by operators instead of installed from manifests managed by the CVO?
+
+### Test Plan
+
+In order to claim full support for this configuration, we must have
+CI coverage informing the release.
+An end-to-end job using the bootstrap-in-place installation flow,
```

Comment:
@eranco74 I think it's already different, right?

Reply:
maybe we also should elaborate here that it will be different CI from the SNO payload that will run with a bootstrap vm on aws/gcp.